### DISCUSSION TITLE
FlatLaf not appearing as an option in a new Netbeans Platform application.

### DISCUSSION THREAD

**Discussion by dat-linux** [apache/netbeans#5921] [Q&A] [answered]
I have NB17 on linux. The IDE itself has the newer flatlaf look and is fine. But upon creating a new NB **Platform Application** (without any changes, just a brand new project), it starts in the old Metal LaF, and only shows the other 3 standard LaF options in the settings. There is no setting option shown for Flatlaf light or dark. The project dependencies have the libraries activated.

I'd like for Flatlaf Light to not just be an option, but set as the default as well. (beyond that I'd like to use the custom Flatlaf theme that NB itself uses).

From what I've read elsewhere, some are saying flatlaf comes up as an option as standard (even if not set as default), so something here seems a bit strange.

Any help appreciated. (this is not about the IDE but creating and then running platform (RCP) applications). I tried in Windows as well, platform apps do not provide FlatLaf option at all.

**Comment by neilcsmith-net** [comment]
Check out #5047 and #3332 and in particular https://github.com/apache/netbeans/pull/3332#issuecomment-979442853

**Reply by dat-linux** [reply]
Nothing in those posts solves the problem I posted.

"The same way that the IDE does this - branding. "

An example would be nice. The changes shown in #3332 relate to the IDE, which as I stated has no issues. I need to add flatlaf as a settings option to a platform application, and preferably have it load as default.

**Reply by neilcsmith-net** [reply, marked as answer]
> The changes shown in https://github.com/apache/netbeans/pull/3332 relate to the IDE, which as I stated has no issues.

The IDE is a platform application!

Please see @oyarzun answer, as just testing with the basic wizard, this shows up as a look-and-feel option correctly using RELEASE170.

To make it show as the default, follow the same process as #3332 - create a file inside the -branding module, creating all the subfolders, at -

`nbm-branding/modules/org-netbeans-swing-plaf.jar/org/netbeans/swing/plaf/Bundle.properties`

Add one line to the file -

`LookAndFeelClassName=com.formdev.flatlaf.FlatLightLaf`

Rebuild the -parent project and run the application.

**Comment by oyarzun** [comment]
Is it a maven project? It sounds like you may have chosen an older
platform version when creating the project. What is the netbeans platform
version in the pom?

    <properties>
        <netbeans.version>RELEASE170</netbeans.version>
        <brandingToken>testlaf</brandingToken>
    </properties>
